### PR TITLE
Add new exercise: Anagram

### PR DESCRIPTION
**Pull request by rchavarria**
Just a new exercise.

### CODE REVIEW

**Comment by kytrinyx** [comment]
Thanks!

File: anagram/anagram_test.spec.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  xit("detects anagrams case-insensitively", () => {
+    let subject = new Anagram("Orchestra");
+    let matches = subject.matches(['cashregister', 'Carthorse', 'radishes']);
+
+    expect(matches).toEqual(['Carthorse']);
+  });
+
+  xit("does not detect a word as its own anagram", () => {
+    let subject = new Anagram("banana");
+    let matches = subject.matches(['Banana']);
```
Comment (iamvery) [inline anagram/anagram_test.spec.js]: Curious, what is the intended behavior if the cases match. For e.g. `subject.matches(['banana'])`. Would you intend `expect(matches).toEqual([])` here as well? I'm assuming so based on the test description, but I don't think that case is covered. What do you think?

Comment (iamvery) [inline anagram/anagram_test.spec.js]: Ah forgive my ignorance, I just realized that the combination of this test w/ the one before it _does_ cover that case. Apologies!

Comment (kytrinyx) [inline anagram/anagram_test.spec.js]: @iamvery 👍

We have a canonical data set that tracks are welcome to (but not obliged to) follow: https://github.com/exercism/problem-specifications/blob/master/exercises/anagram/canonical-data.json#L80-L100

**Pull request merged**